pkg/builder/local_build_executor.go

```diff
@@ -248,6 +253,24 @@ func (be *localBuildExecutor) Execute(ctx context.Context, filePool re_filesyste
 		return response
 	}
 
+	if len(be.inputRootCharacterDevices) > 0 {
+		if err := inputRootDirectory.Mkdir("dev", 0777); err != nil && !os.IsExist(err) {
+			attachErrorToExecuteResponse(response, util.StatusWrap(err, "Unable to create dev directory in input root"))
+		}
+		devDir, err := inputRootDirectory.EnterDirectory("dev")
```

Comment:
This code currently leaks `devDir`. If it makes your life easier, just add a helper function to `localBuildExecutor`, so you can use `defer devDir.Close()` there to close the directory afterwards.